src/clarity-angular/datagrid/providers/selection.spec.ts

```diff
@@ -135,5 +141,95 @@ export default function(): void {
             this.selectionInstance.currentSingle = 2;
             expect(nbChanges).toBe(3);
         });
+
+        it("clears selection when a filter is added", function() {
+            this.selectionInstance.selectionType = SelectionType.Multi;
+            this.selectionInstance.current = [4, 2];
+
+            let evenFilter: EvenFilter = new EvenFilter();
+
+            this.filterInstance.change.subscribe(() => {
+                expect(this.selectionInstance.current.length).toBe(0);
+            });
```

Comment:
I don't like this pattern, because if someone removed the filter listener in the Selection provider, this test would _always_ pass, even if it didn't clear anymore.

I think you just want the `expect()` to go right after the `this.filterInstance.add(evenFilter);`, you don't need to subscribe to `.change` in the test.